xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/skin/SkinEnvironmentResource.java

```diff
@@ -49,6 +55,13 @@ public SkinEnvironmentResource(String path, String resourceName, ResourceReposit
     public String getURL(boolean forceSkinAction) throws Exception
     {
         XWikiContext xcontext = this.xcontextProvider.get();
+
+        URL resourceUrl = this.xcontextProvider.get().getEngineContext().getResource(this.getPath());
+        Map<String, String> parameters = new LinkedHashMap<>();
+        Path resourcePath = Paths.get(resourceUrl.toURI());
```

Comment:
You should probably bulletproof a bit this line in case this resource cannot be converted to a local file for some reason (like not expended WAR or weird URL scheme like in WebSphere).

Reply:
ok I'll try/catch them.

Reply:
Should probably also fallback on the XWiki version if there is an issue.

Reply:
Same comments for the duplicated code located in XWiki.java (which should probably me moved in some common place, ultimately what's in XWiki.java right now should probably be refactored into a something extending AbstractEnvironmentResource but that's for later).